homeassistant/components/launch_library/sensor.py

```diff
@@ -1,77 +1,83 @@
-"""A sensor platform that give you information about the next space launch."""
+"""Support for Launch Library sensors."""
 from __future__ import annotations
 
-from datetime import timedelta
-import logging
+from typing import Any
 
-from pylaunches import PyLaunches, PyLaunchesException
-import voluptuous as vol
+from pylaunches.objects.launch import Launch
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from .const import (
     ATTR_AGENCY,
     ATTR_AGENCY_COUNTRY_CODE,
     ATTR_LAUNCH_TIME,
     ATTR_STREAM,
     ATTRIBUTION,
-    DEFAULT_NAME,
-)
-
-_LOGGER = logging.getLogger(__name__)
-
-SCAN_INTERVAL = timedelta(hours=1)
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}
+    DOMAIN,
 )
 
 
-async def async_setup_platform(
```

Comment:
This still needs to be present for 2 releases.
The platform setup needs to trigger an import flow, and warn the user with a log entry to remove their configuration.